### PR TITLE
test_all_sandia,update_lib: remove unsupported compilers

### PR DESCRIPTION
**Pull request by ndellingwood**
add module workaround for intel/17

### CODE REVIEW

**Comment by ndellingwood** [comment]
Updates for c++14 minimal required standard

**Comment by dalg24** [comment]
Please look over https://github.com/kokkos/kokkos/tree/7889cad2e6f845912d548f04fcf039b39195b6d2#primary-tested-compilers-on-x86-are and update if necessary

File: scripts/testing_scripts/test_all_sandia [hunk 1]
```diff
@@ -508,10 +497,7 @@ elif [ "$MACHINE" = "apollo" ]; then
 
   if [ "$SPOT_CHECK" = "True" ]; then
     # Format: (compiler module-list build-list exe-name warning-flag)
-    COMPILERS=("gcc/4.8.4 $BASE_MODULE_LIST "OpenMP,Pthread" g++ $GCC_WARNING_FLAGS"
-               "gcc/5.3.0 $BASE_MODULE_LIST "Serial" g++ $GCC_WARNING_FLAGS"
-               "intel/16.0.1 $BASE_MODULE_LIST "OpenMP" icpc $INTEL_WARNING_FLAGS"
-               "clang/3.9.0 $BASE_MODULE_LIST "Pthread_Serial" clang++ $CLANG_WARNING_FLAGS"
+    COMPILERS=("gcc/5.3.0 $BASE_MODULE_LIST "OpenMP,Pthread,Serial" g++ $GCC_WARNING_FLAGS"
```
Comment (jeffmiles63) [inline scripts/testing_scripts/test_all_sandia:500]: in updating the read me, I saw that we removed selective version of clang...on apollo, clang3.9.0 uses gcc 5.3.0 and clang says that c++14 is supported with 3.4 or later.  Thus, I'm not sure if we were supposed to remove clang 3.9.0

File: scripts/testing_scripts/test_all_sandia [hunk 2]
```diff
@@ -524,16 +510,9 @@ elif [ "$MACHINE" = "apollo" ]; then
...
-               "clang/3.9.0 $CLANG_MODULE_LIST $BUILD_LIST_CLANG clang++ $CLANG_WARNING_FLAGS"
-               "gcc/4.8.4 $BASE_MODULE_LIST $GCC_BUILD_LIST g++ $GCC_WARNING_FLAGS"
-               "gcc/4.9.3 $BASE_MODULE_LIST $GCC_BUILD_LIST g++ $GCC_WARNING_FLAGS"
                "gcc/5.3.0 $BASE_MODULE_LIST $GCC_BUILD_LIST g++ $GCC_WARNING_FLAGS"
                "gcc/6.1.0 $BASE_MODULE_LIST $GCC_BUILD_LIST g++ $GCC_WARNING_FLAGS"
-               "intel/15.0.2 $BASE_MODULE_LIST $INTEL_BUILD_LIST icpc $INTEL15_WARNING_FLAGS"
-               "intel/16.0.1 $BASE_MODULE_LIST $INTEL_BUILD_LIST icpc $INTEL_WARNING_FLAGS"
                "intel/17.0.1 $BASE_MODULE_LIST $INTEL_BUILD_LIST icpc $INTEL_WARNING_FLAGS"
-               "clang/3.5.2 $BASE_MODULE_LIST $CLANG_BUILD_LIST clang++ $CLANG_WARNING_FLAGS"
-               "clang/3.6.1 $BASE_MODULE_LIST $CLANG_BUILD_LIST clang++ $CLANG_WARNING_FLAGS"
```
Comment (jeffmiles63) [inline scripts/testing_scripts/test_all_sandia:536]: same as comment above, but it looks like clang 3.6.1 uses gcc 5.3.0
clang 3.5.2 uses gcc 4.7.2 so it is probably ok to remove

File: scripts/testing_scripts/update_lib.sh [hunk 3]
```diff
@@ -1,6 +1,33 @@
 #!/bin/bash
 
 if [ "$1" = bowman ]; then
+  ICPCVER="$(icpc --version | grep 17 | cut -d ' ' -f 3)"
+  if [[ "${ICPCVER}" = 17.* ]]; then
```
Comment (jeffmiles63) [inline scripts/testing_scripts/update_lib.sh:5]: I think this needs to be for 17 and 18.  18 looks like the default is 4.9.3 as well

File: scripts/testing_scripts/test_all_sandia (hunk 1 above)
Comment (dalg24) [reply scripts/testing_scripts/test_all_sandia:500]: Good catch.  Looking at the [developer meeting notes on the internal repo](https://github.com/kokkos/internal-documents/blob/master/meeting-notes/2020-08-12.md) we decided we'd drop support for the 3.x series.  So Nathan's changes are right.

File: scripts/testing_scripts/update_lib.sh (hunk 3 above)
Comment (ndellingwood) [reply scripts/testing_scripts/update_lib.sh]: Thanks for pointing that out, just pushed an update

Comment (ndellingwood) [reply scripts/testing_scripts/update_lib.sh]: Made similar update for blake, 18.0 loads gcc/4.9.3 but 18.1 does not.

**Comment by masterleinad** [comment]
Retest this please.

**Comment by dalg24** [comment]
Re-running the test was pointless

**Comment by ndellingwood** [comment]
> Re-running the test was pointless

@dalg24 @masterleinad I forgot to add [skipci], sorry

**Pull request merged**